# Text and Tables Extraction on ICLR Data - PART2

This is for the references of ICLR papers

In [2]:
import pickle
from collections import defaultdict
import re
import glob

In [3]:
from pathlib import Path
from axcell.helpers.paper_extractor import PaperExtractor
from axcell.data.paper_collection import PaperCollection
from axcell.models.structure import TableType, TableStructurePredictor, TableTypePredictor
# from axcell.helpers.results_extractor import ResultsExtractor

## Load the paper references keys for steering

In [32]:
with open('iclr_ref_arxiv_map.pkl', "rb") as f:
    iclr_ref_arxiv_map = pickle.load(f)

## Run extraction one by one 

In [5]:
ROOT_PATH = Path('data')
MODELS_PATH = Path('models')

SOURCES_PATH = ROOT_PATH / 'sources'
PAPERS_PATH = ROOT_PATH / 'papers'

In [34]:
extract = PaperExtractor(ROOT_PATH)

In [35]:
models_path = Path(MODELS_PATH)
ttp = TableTypePredictor(models_path, "table-type-classifier.pth")
pc = PaperCollection.from_files(PAPERS_PATH)

/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/torch/serialization.py:453: SourceChangeWarning: source code of class 'torch.nn.modules.loss.BCEWithLogitsLoss' has changed. you can retrieve the original source code by accessing the object's source attribute or set `torch.nn.Module.dump_patches = True` and use the patch tool to revert the changes.
  warnings.warn(msg, SourceChangeWarning)
/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/torch/serialization.py:453: SourceChangeWarning: source code of class 'fastai.text.models.awd_lstm.AWD_LSTM' has changed. you can retrieve the original source code by accessing the object's source attribute or set `torch.nn.Module.dump_patches = True` and use the patch tool to revert the changes.
  warnings.warn(msg, SourceChangeWarning)
/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/torch/serialization.py:453: SourceChangeWarning: source code of class 'torch.nn.modules.sparse.Embedding' has 

In [36]:
table_labels = {TableType.SOTA: 'leaderboard', TableType.ABLATION: 'ablation', TableType.IRRELEVANT: 'irrelevant'}

In [37]:
status_count = 0
err_count = 0
unextracted_papers = []
err_paper_ids = []
unextracted_paper_ids = []
status_count = max(len(leaderboard_table_refs), len(leaderboard_refs))
err_count = 0
errs_ids = []

leaderboard_table_refs = defaultdict(dict)
leaderboard_refs = defaultdict(dict)

In [38]:
already_computed = list(set(list(leaderboard_table_refs.keys()) + list(leaderboard_refs.keys())))

In [39]:
len(already_computed), already_computed[0:3], status_count, err_count, len(unextracted_papers)

(0, [], 5, 0, 0)

In [33]:
total = len(iclr_ref_arxiv_map)
count = 0

# First extract all - i.e. unpack and load htmls
for k, v in iclr_ref_arxiv_map.items():
    count += 1
    if count%10==0:
        print(count, total)
    try:
        path_split = v.split(".")
        # Extraction
        extract(SOURCES_PATH / path_split[0] / v)
    except Exception as ex:
        print(k, v)

Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
647191 1712.04248
Unpacked successfully
html conversion successful
Unpacked successfully
LaTeXML was unable to convert source code of this paper
10 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
20 2328
Unpacked successfully
html conversion successful
Unpack

html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
180 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
190 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
ht

/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "cs_0209010" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
220 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
230 2328
Unpacked successfully
html conversion successful
Un

/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "hep-ph_0702172" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')
/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "hep-ex_0507097" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
300 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
LaTeXML was unable to convert source code of this paper
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conv

/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "math_9905121" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


214306 math_9905121
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
LaTeXML was unable to convert source code of this paper
763177 gr-qc_0404079
Unpacked successfully


/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "gr-qc_0404079" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
LaTeXML was unable to convert source code of this paper
380 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
390 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful


/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "cmp-lg_9407020" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


1398 cmp-lg_9407020
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
400 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
410 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conv

/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "math_0612104" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


787948 math_0612104
430 2328
Unpacked successfully
LaTeXML was unable to convert source code of this paper
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
LaTeXML was unable to convert source code of this paper
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
440 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully


Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
LaTeXML was unable to convert source code of this paper
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
600 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
610 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully


/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "physics_9806030" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


608552 physics_9806030
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
640 2328
Unpacked successfully
LaTeXML was unable to convert source code of this paper
1059134 cmp-lg_9606011


/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "cmp-lg_9606011" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
650 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
660 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Un

/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "cs_0009008" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


1132974 cs_0009008
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
680 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
LaTeXML was unable to convert source code of this paper
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
690 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unp

/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "hep-ph_9704332" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


674625 hep-ph_9704332
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
790 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
800 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
LaTeXML was unable to convert source code of this paper
Unpacked successfully
html conversion successful


/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "hep-th_9707250" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


412794 hep-th_9707250
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
870 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
880 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html co

/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "astro-ph_0404546" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


61052 astro-ph_0404546
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1030 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
LaTeXML was unable to convert source code of this paper
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful


/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "cs_0408058" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


716471 cs_0408058
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1040 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
LaTeXML was unable to convert source code of this paper
1050 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
Unable to find any suitable tex file
Unpacked successfully
html conversion suc

/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "quant-ph_0204007" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


536551 quant-ph_0204007
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1140 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1150 2328
Unpacked successfully
html conversion successful
Unpacked successfully
htm

/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "cond-mat_9705270" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


545116 cond-mat_9705270
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful


/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "math_0409267" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


315758 math_0409267
1240 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1250 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful


/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "cond-mat_0406368" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


250237 cond-mat_0406368
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1260 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1270 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
htm

/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "physics_0603229" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


389591 physics_0603229
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1290 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1300 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html

/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "cs_9301114" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


955016 cs_9301114
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1340 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful


/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "physics_0601009" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


160452 physics_0601009
1350 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1360 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
LaTe

/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "physics_9803008" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
LaTeXML was unable to convert source code of this paper
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1390 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
LaTeXML was unable to convert source code of this paper
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1400 2328
Unpacked succ

/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "cs_0306050" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


423140 cs_0306050
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1510 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1520 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful


/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "quant-ph_0507189" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


103720 quant-ph_0507189
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1530 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
LaTeXML was unable to convert source code of this paper
1540 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion success

/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "cond-mat_9910332" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')
/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "math_0501398" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


472399 cond-mat_9910332
1045843 math_0501398
Unpacked successfully
html conversion successful
1550 2328
Unpacked successfully
LaTeXML was unable to convert source code of this paper
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1560 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
ht

/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "hep-ph_0512082" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


357802 hep-ph_0512082
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1580 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1590 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html 

/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "physics_0609135" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


1038045 physics_0609135
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1660 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1670 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
htm

/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "hep-th_9908205" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


907099 hep-th_9908205
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1700 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1710 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html 

/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "hep-th_9510025" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


587854 hep-th_9510025
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1770 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
LaTeXML was unable to convert source code of this paper
Unpacked successfully
html conversion successful
Unpacked successfully
LaTeXML was unable to convert source code of this paper
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1780 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfu

No article found for 1901.00544


Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1800 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1810 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful


/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "cs_0409058" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


473448 cs_0409058
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1850 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1860 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conv

/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "math_0507204" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


571973 math_0507204
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1920 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1930 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html co

/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "physics_0509039" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
LaTeXML was unable to convert source code of this paper
Unpacked successfully
html conversion successful
1950 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
1960 2328
Unpacked successfully
html conversion succ

/home/singh_shruti/anaconda3/envs/axcell/lib/python3.7/site-packages/axcell/helpers/paper_extractor.py:26: UserWarning: Unable to infer arxiv_id from "cs_0506075" filename
  warnings.warn(f'Unable to infer arxiv_id from "{source.name}" filename')


801979 cs_0506075
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
2110 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
LaTeXML was unable to convert source code of this paper
2120 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Un

Unpacked successfully
html conversion successful
2270 2328
Unpacked successfully
html conversion successful
Unpacked successfully
LaTeXML was unable to convert source code of this paper
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
LaTeXML was unable to convert source code of this paper
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
2280 2328
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
html conversion successful
Unpacked successfully
LaTeXML was unable to convert source code of this paper
Unpacked successfully
LaTeXML was unable to convert source c

In [40]:
count, total

(2328, 2328)

In [41]:
for k, v in iclr_ref_arxiv_map.items():
    if not (k in already_computed or k in err_pape0==r_ids or k in unextracted_paper_ids):
        if v:
            if status_count % 10 == 0:
                print("Status: ", status_count)
                with open("REF_leaderboard_table_refs.pkl", "wb") as some_file:
                    pickle.dump(leaderboard_table_refs, some_file)
                with open("REF_leaderboard_refs.pkl", "wb") as some_file:
                    pickle.dump(leaderboard_refs, some_file)
                    
            try:
                path_split = v.split(".")

                # Extraction
                # extract(SOURCES_PATH / path_split[0] / v)

                paper = pc.get_by_id(v)
                
                if not paper:
                    unextracted_papers.append((k, v))
                    continue
                
                print(k)
                
                if len(paper.tables) > 0:                    
                    # Add table type if absent
                    if paper.tables[0].gold_tags == "":
                        tables_types = ttp.predict(paper, paper.tables)
                        for table, table_type in zip(paper.tables, tables_types):
                            table.gold_tags = table_labels[table_type]
                    
                    #Count the different types of tables
                    table_type_count = [0, 0, 0]
                    for tt in paper.tables:
                        if tt.gold_tags.lower().strip() == "leaderboard":
                            table_type_count[0] = table_type_count[0] + 1
                        elif tt.gold_tags.lower().strip() == "ablation":
                            table_type_count[1] = table_type_count[1] + 1
                        elif tt.gold_tags.lower().strip() == "irrelevant":
                            table_type_count[2] = table_type_count[2] + 1
                        else:
                            print("Unknown table type: ", tt.gold_tags)
                    
                    # Save table stat in leaderboard_table_refs
                    leaderboard_table_refs[k]["tcount"] = [len(paper.tables), table_type_count[0], table_type_count[1], table_type_count[2]]
                else:
                    leaderboard_table_refs[k]["tcount"] = [0, 0, 0, 0]
                
                leaderboard_table_refs[k]["table_labels"] = []
                for tablerol in paper.tables:
                    leaderboard_table_refs[k]["table_labels"].append((tablerol.gold_tags,))
                
                if (leaderboard_table_refs[k]["tcount"][1]+leaderboard_table_refs[k]["tcount"][2]) > 0:
                    leaderboard_refs[k]["refs"] = {}
                
                # If sota/leaderboard tables present, extract references from them
                if leaderboard_table_refs[k]["tcount"][1] > 0:
                    leaderboard_refs[k]["refs"]["ldb"] = []
                    leaderboard_refs[k]["count"] = 0
                    for iterid, table in enumerate(paper.tables):
                        idx = iterid + 1
                        if table.gold_tags == "leaderboard":
                            
                            table_dir_path = "/home/singh_shruti/workspace/axcell_ws/axcell/notebooks/data/papers/" + path_split[0] + "/" + v
                            table_files = glob.glob(table_dir_path+"/*.csv")
                            table_files_stripped = [tfs.rsplit("/", 1)[1] for tfs in table_files]
                            
                            candidate_name = "table_" + "{:02d}".format(idx) + ".csv"
                            if candidate_name in table_files_stripped:
                                cannot_use_f = open(table_dir_path + "/" + candidate_name, "r")
                                for line in cannot_use_f:
                                    m = re.findall("<ref id=[0-9a-zA-Z'-]*>[a-zA-Z 0-9\.]*</ref>", line)
                                    for iim in m:
                                        leaderboard_refs[k]["refs"]["ldb"].append(iim)
                                        leaderboard_refs[k]["count"] += 1
                            else:
                                print("For {} cannot find leaderboard table file: {}".format(k, candidate_name))
                            
                    leaderboard_refs[k]["refs"]["ldb"] = list(set(leaderboard_refs[k]["refs"]["ldb"]))
                    
                if leaderboard_table_refs[k]["tcount"][2] > 0:
                    leaderboard_refs[k]["refs"]["abl"] = []
                    for iterid, table in enumerate(paper.tables):
                        idx = iterid + 1
                        if table.gold_tags == "ablation":
                            
                            table_dir_path = "/home/singh_shruti/workspace/axcell_ws/axcell/notebooks/data/papers/" + path_split[0] + "/" + v
                            table_files = glob.glob(table_dir_path+"/*.csv")
                            table_files_stripped = [tfs.rsplit("/", 1)[1] for tfs in table_files]
                            
                            candidate_name = "table_" + "{:02d}".format(idx) + ".csv"
                            if candidate_name in table_files_stripped:
                                cannot_use_f = open(table_dir_path + "/" + candidate_name, "r")
                                for line in cannot_use_f:
                                    m = re.findall("<ref id=[0-9a-zA-Z'-]*>[a-zA-Z 0-9\.]*</ref>", line)
                                    for iim in m:
                                        leaderboard_refs[k]["refs"]["abl"].append(iim)
                            else:
                                print("For {} cannot find ablation table file: {}".format(k, candidate_name))
                            
                    leaderboard_refs[k]["refs"]["abl"] = list(set(leaderboard_refs[k]["refs"]["abl"]))
                            
                status_count += 1
            except Exception as ex:
                print(k, v)
                print("Error: ", ex)
                err_count += 1
                errs_ids.append((k, v))

1179651


1179653


598027
1163279


557076
Status:  10
983062


647191


557082


229423


172080


622648


319548


327741


622657


1138755


Status:  20
1155139


155719


41035


737358


1171539
1163347


540762
1187930


655452


106587


Status:  30
82026


1187952


155762


1163380


1081461


385142


1171574


For 1171574 cannot find leaderboard table file: table_02.csv
For 1171574 cannot find leaderboard table file: table_04.csv
For 1171574 cannot find ablation table file: table_03.csv
For 1171574 cannot find ablation table file: table_05.csv
868479


368768


868481
Status:  40
32898


368771
860290


139398


1056902


1155208


958605


1147022


1122449


163988
Status:  50
1155227
180383


73894


1171624


1040553


1155242


1007786


1196205


1171634


860339
Status:  60
229558


229559


1015990


123068


1032383


1138883


827588


139462


65737


942286


Status:  70
1163474


16595


409813


483547


106720
671972
1155304
123118


581874


147699


Status:  80
680181


1155319


1204477


712958


581890


811269


246024


606473


753966


1155377


Status:  90
1155378


1147188


147768


1188156


942397


1048899
1155408


639320


1171805


418147


Status:  100
Status:  100
147814


115054


1155439


778609


1155442


557426


377208


1155449


688514


958851


Status:  110
Status:  110
Status:  110
Status:  110
942483


1016213


1188264


328106
467373


401847


205240


205248
704964


590277


Status:  120
1155524


672196


1024457


360922


868828


16861


1163743
664032


1122786
1171939


Status:  130
475630


1188338


1163769
1155584
205313


1171968


705037


696851
1171988


1188375


Status:  140
1040928


1180192


696866


1180195


1180202


213553
582204


98878


270911
639555
Status:  150
180805
295495


270928


1180242


336470


492122


287334
1114729


1016426


819818


Status:  160
1180268


107113
950895


148079


1155696


1172082


959095


819832


254587


1180292


Status:  170
279176
819849


590474


164493


1180313


926365


459424
1172129


1188519


1155754


Status:  180
524972


574126
533169


1163954


901811
901809


99004


1172158


615105


1082054
Status:  190
1155786


1180363


1188558


582358


877274


312027


1155803


1114846


1155826


451317


Status:  200
Status:  200
131843


262920
647946


738059


1032972


181005


353047


647960


910107


1008422


Status:  210
959271


1008425


1008429


992046


361261


1131319


303930


1073979


1164092


779070


Status:  220
Status:  220
1131332


312131


598854


279371
574288


705370


181085


820065


680802


615267


Status:  230
844647


598892


1065838


516983


1164151


222073


1082252


1090447


91025


713621


Status:  240
25502


877475


1172387


1098664
730027


877483


1098667


918444


1180587


394160


Status:  250
Status:  250
1008572


517053


115646


1180605


41925


1147846


213959


754630


1156042


361420


Status:  260
705486


558030


934865


336850


566227


1180628


943067
975842


1180647


1180648


Status:  270
615402


58348


533486
1164273


869362


1172474


1164284


1172478


861194


459788


Status:  280
730127


1180689


1164311


1066018


394276


582693
99367


943144
500773


50221
Status:  290
1164337


1164346


951359


640065


549955


746563


1180742


123980


1180752
1180756


Status:  300
173148


672869


1135


25713


1148018


500851


926835


1123447
656510


353409


Status:  310
Status:  310
74887


722056


255117
591002


558235
1082523


1057948
500898


959652


1164452


Status:  320
1156260
1180840


42157


148657


1189048


1164473


91323
926908


91327


148676


Status:  330
1172681


386255


1180880


1066194


247001
247004


492765


214237
1164510


1090796


Status:  340
Status:  340
967925
632066


206085


951561


140560


1107219


632085
1000730


1156381
99616
Status:  350
Status:  350
83235


984358


75065


681279


1017161


476491


976204


361808


681303


1090905


Status:  360
1189211


1172828


1082720


329057


1181025


836972


197998


1181040


542069
591234


Status:  370
1172867
107908


853382


1172871


992655


664975


1164689


1058196


1189273
419232


Status:  380
386466


1172898


1156522


214442


894386


746931


1156537
984509
107965


959946
Status:  390
Status:  390
746961
1172946


1091029


107989


550361


599515


779745


353768


386538


804330


Status:  400
Status:  400
Status:  400
1099256


1148411
288253


165373


370175
894464


1181186


1164806


1181190
1164822


Status:  410
951831


755224
1156638


1189407


1156640


443937


1189411


1173035
1173038


714297


Status:  420
1181248


1601


1164866


443971


1140292


820810
1164882


9814
247383


542297


Status:  430
1181276


1173086


222816


222817
837218


1156712


747118
99956


673398


550519


Status:  440
231034


632446


1132159


1189506


181898


509578


1017485


525973


657047


222872


Status:  450
190105


1164955


75420
878245


1156779


1181363


648886


468663
124599


542395


Status:  460
362175


886465


534213


34514
542420


1124057
1066714


943835


1132254
722660


Status:  470
1165031


747242


1173229


894715


222979
345860
395027


263958


124697


83737


Status:  480
1140513


67365


616233


370473


763691
1034028
362287


50998
1156928


1156930


Status:  490
272195
1107779


190284


1156940


132947
1066836
1156948


608094


575327


993122


Status:  500
Status:  500
804721


608123


1165190


886663
493449


1173388


1058704


157588


1173400


313249
Status:  510
18341


870311


960425


198571
190391


1189816


829377


911311


1173456


42969


Status:  520
1157081


1173468


1140712
542699


1181678


1165307


321533


1181695


960513


731146


Status:  530
223243


690187
632851


1189911


1173534


1173537


1181732


43059


1001527


247864


Status:  540
862265


829498


84038


1149002


1157197


575568
1165396


2133


1157206


1173594


Status:  550
125022


723041
960613
551035


551044
1165460


444569
1083551


1181860


452773
Status:  560
977062


452778


510128


944307


59572


1018037


288951


1181891


116956


796897
Status:  570
1190115


1157347


682212


780525


1149167


174321


207090


108787


108788


821500


Status:  580
829692


289026


993543


608523


10514
1181972


59672


149788


1165601


1165607


Status:  590
1149224


215337


108849


1157426


26933


977205


1182015
182601
1173835


207180


Status:  600
829776


92501


108888


518490


100702


534882


592228


1050981


1026412


1034612


Status:  610
100725


264565


1165688


862590


1173887


67969


84357


1149321
1157513


1173910


Status:  620
534935


59804


1124765


674208


772519


600488


1067443


1190324


1165747


1149368
Status:  630
215484


952767


387523


903621


27080


1182154


1124812


666061


338383


928211


Status:  640
756182


338393


887261


1157598
1165791


944617


707049


109036


575986


543229


Status:  650
748040


1165833


510474


346635
887309


174609


1157661


1190430


469540


494118


Status:  660
748084


1100340


830007


920120


363067
739903


19008


518723
928323


1157704


Status:  670
739916


1182286
100944


289376


461415
1149554
739956


903807


895618


1125001
Status:  680
903822


1092245


182936


993944


592545


117412


158373


789158


371368


182953


Status:  690
780970


125627


84668


92862


658122


1174219


1166026


371406


838353


232146
Status:  700
666324
772825
469726


273121


166628


2788
1149672


68336


862960


502521


Status:  710
445182


1182465


174853


1043205


723719


1149705


305932


1174284


772879
961304


Status:  720
35611


1174305


68386


215843


477987


256805


371495


1157933


928575


1157954
Status:  730
1174340


494406


330568


936778


191307


2895
60242


985945


133978
314203
Status:  740
84828


1174370


1182566


674664
1157995


1117042


281462


281463


150394


838523


Status:  750
Status:  750
510846


1133440


101253
625542


674697
871310


60318


633759


617383


1174439


Status:  760
1174441


592819


560055


322491


199615


625600


1092547


166866


658388


216031


Status:  770
805856


1190888


1190892


461822


1199103


928778


502800


1149973


1174552


592927


Status:  780
1051681


1158177


994344


1158187
724014


404528


1035319


1158206


1158210


265286


Status:  790
1150024
1043530
953422


1125463


470104


945254


502887


978024


1158257


683125


Status:  800
822391


527483


1191042


1166469


1150088


1182858
1084555


1150091


1158287


879773
Status:  810
1158301


494751


289954


76965


314533
1182887


1182885


1092780


265390


1150128


Status:  820
1002672


756922


1084605


1158338


1166534


330961


167127


1150171


1182939


167133
Status:  830
281823
634081
789730
642279
912623


412911
355574


1158406


503047


290060
Status:  840
1084693


986391


1084696


1158436


642342


986408


699695


1183024


1150259


347446


Status:  850
937273


970047


896321


1150279


683338


1158475


1174858


1183053


929104


1166673


Status:  860
1174866


970071


789850


961885


1109346
1183079


1133929


568683


920951


691587


Status:  870
1002883


748937


421259


1150347


601486


19856


1084817


1150353


36246


1150361


Status:  880
298397


691617


208292


1158564


871848


871850


77231


765360


519599
1150388


Status:  890
380342
781750
585148


994749
1158599


1183177


380363


331213


519629


191953


Status:  900
462303


1142241


1076712


1076716
1166832


191991
224763


880124


519676


577022


Status:  910
642561


224770


814606


1150478
609808


1134096
1084947


159255
364056


994839
Status:  920
888347


962076


1166875


224801


405026


1175075


831019


1044012


429615


1191477


Status:  930
142903


601660


552508


224836


1150535


1150537


847439
3663


1175124


691799


Status:  940
921175


1150564


20074


167534


790126


1166962
314995


102007


601723


1158780


Status:  950
1175163


904846


683673


1183389


1167006


315043


470697


36525


1183413


1183419


Status:  960
1183420


1019581


1068740


831177


290511


1150676


806620
1183453


618221
700141


Status:  970
Status:  970
380656


339697


855792
159475


347891


888571


511760
429843


1142551


1167127


Status:  980
36639


880417


511778
421670


577321
298793
905004
85806


839470


839477


Status:  990
667446
1158970


1150779


1158972


372541
290620


765774


1191758


1183566


716627
Status:  1000
1191767


1150810


577373


1159006


790368


1191780


1175398


692073


470891


675695


Status:  1010
1159025
1191796
741237


3958


233336
855941


864137
1175453


1159071


421794
Status:  1020
1183651


815013


339878


200616


765865


1167275


1134509


798638


823216


446387


Status:  1030
708536


372668


1142720


200646


1150925


724946


577490


200660


1044436


675798


Status:  1040
249811


774108


61410


733154


290786
356332


290797


1167345


28660


536567


Status:  1050
266236


1060862


1159169


806915


880644
331781


364551


446477
700441
528410


Status:  1060
471066


1175580


1183773


479267


1159208


692266


593965


1159219


1142841


380993


Status:  1070
1019975


1175627


921676


454733


143440


995417


1151065


1151068


831592
127082


Status:  1080
1151085


643184


643188
86134


643193


864380
1159314


1069205


1183903


1183907
Status:  1090
1183909


282792


184492


274609


1192117


807095


430263
1028281


1020088


102585


Status:  1100
1167543


1167560
1151179


962765


372941


1142991


430288


1159382


1167577


856282


Status:  1110
1159391


1020130
921830


1183976


1183989


159990


577783


1167606


258295


995578


Status:  1120
1159419


1151228


1159425


659731


1151252


1159448


45337


168220


889118


848159


Status:  1130
86307


864554


725292


1151279


807219


1167668


168245


332087


807224


160058


Status:  1140
364861


1159485


905550


905552


536921


1159517


512350
594284


758126
1167728


Status:  1150
1102193


1167727


938355


708977


627061
1151342


946551


897402
479611
1167739


Status:  1160
1110397


766335


1143168


1159549


815494


1069447


1069454


569749


45461


1151383


Status:  1170
1061275


1151393


913828


414117
954800


823730


29107


422324


537018


577981
Status:  1180
577982


504257


1176002


274884


823752


651722


1167826
1151443


1184211


446935


Status:  1190
Status:  1190
840160


430561


922082


651749


1176038


815591


1159654


586218


1167851


741871


Status:  1200
1102326


840189


1061373


307720


938506
1053206


78359


1192471


987677


545309
Status:  1210
553506


1151522


528933


414250


1110572


447022


1167922


668221


78399


512579


Status:  1220
1151558


275014


283209


1159754


315981
840271


1159763


782939


1012318


471655


Status:  1230
1151591


610927


201329


1159795


1159798


995960


1167993


1176196


914058


135824


Status:  1240
225937


1192596


717465


1184411


496284


1168032
479906


774827


1151667


1151669
Status:  1250
Status:  1250
1168068


1192647


758472


45773
94925
1168086


1168092


529117


1159900


45795


Status:  1260
1176294


111334


176873


914156


791282


627442


1159926


127737


1159929


332551


Status:  1270
160522


168717


1176337


1086226


815892


824092


160547


1159976


267049


1176363


Status:  1280
316210


1159991


1053506


455492
1135436


1184588


611151


578386


1176404


1151840


Status:  1290
Status:  1290
635748


1201001


1184618


1151863


1184637


930686


766849


594817


201608


78729
Status:  1300
897931


1037198


1143697


250773
709528
660378
1176477


537503


1045413


299943


Status:  1310
914344


611241


1160114


1151922


586683


1184701


644032


185281


1160129


1192900


Status:  1320
1160136


37836


685005


758732


1160150


1045463


1176537


742362


291803


1168349


Status:  1330
480222


1127393


1160167


1176551


5098


1168365


644078


1160179


463860
234486


Status:  1340
463867


78845


750596


201733


259081


857100


1168409


1160221


840735


594980


Status:  1350
586791


709672


463915


357428
504890


226366


586820


824392
1184843


791630


Status:  1360
1152079


914512


832594
1168467


668764


1193063


898155


988269


455790


357489


Status:  1370
357490


1168504


1184892


971904


1193089


185487


13457


193682


193683
5268


Status:  1380
406679


218263


21655
1168541


332961


578725


292008


816300


1160377


849083


Status:  1390
1160385


1176775
251082


718033


464084


1160405


46294


513240
619740


111844


Status:  1400
Status:  1400
996585


1176809


46321


955634


1152243


152830
218381


54547


914708


234773


Status:  1410
1078552


734492
87327


595232


1045791
1185059


1176868


1185061


996657


677170


Status:  1420
1152305


726326


30010


1176890


513340


759101
841023


1176896


1160512


1185088


Status:  1430
Status:  1430
267591


1160518


87367


1160523


619852
775502


677199


1160534


800093


111966


Status:  1440
898400


1193319


1152359


1168746


1168748


447867
111996


841085


210309


79238


Status:  1450
710023


1136007
841098


488843
1152415


1160607


46498


1119654


1176998


865705


Status:  1460
Status:  1460
1037741


308656


415153
1160624


1152437


1177014


988598


275898


1160634


939453


Status:  1470
1160642


1160648


1160650


1177035


521676


538061
1177037


398806


464347


390621


Status:  1480
1168874


529900


226804


1062389
1160715
226828


1144335


914975


398884


366120


Status:  1490
947753


497194
366123


1177132


939565


185905


579121


579127


988728


177723


Status:  1500
865857


1168961


1177158


1193557


996950
341590


398936


1160791


415327


1177183


Status:  1510
603747


1046115


46693


357989


1152623
1152625


693874


210556


1177227


947852


Status:  1520
Status:  1520
1095312


1152662


1152676


300712
276140


202422


333505


243394


988869


1185478


Status:  1530
923336


464586


1152722
775890


710363


587498


1160942


366318


308976
1079026
Status:  1540
1013496


1152762


227073
14082


96004
120581
1152776


38666


55050


292621
Status:  1550
1177358


661266


587540


587545


579359


956195


874278


120622


227135


112447


Status:  1560
767809
907079


767816


227145


202572


448333


1169230


1152852


1103704


382811
Status:  1570
1152861


431971


997219


423785


1169258


423787


620399


169840


325489


644977


Status:  1580
1169265


161652


628600


1152894


669569


956294


1070982


112522
825230


563087


Status:  1590
1161105


874386


1152916


292759


1152933


1161129
522155
456620


1177519


1177527


Status:  1600
1169336


300986


1177530


38849


1095618


1161159


489416


374728


1013709
890830


Status:  1610
382930


997330


210901


931799


1152994


530405


858089


104426


1153004


489461


Status:  1620
317434


890877


1153025


473092


1153029


6149
1112078


1136654
1161232


1153043
Status:  1630
1185823


1153061


1153062


1112103


612407


604233


1177678


1177683


1153112
530521


Status:  1640
710746


1153122


1153127
1079402


522347


866413


1054841


333955


1161354


1136781


Status:  1650
227472


1161364


252056
866458


809115


284828


661668


1169578


301244


587968
Status:  1660
1038531


1071300
6346


817355


1013966


1169615
1153239


137434


1153244


202973


Status:  1670
1046751


465128
1161449


1177833


669934


514289


1079538


1153267


1186044


1153289


Status:  1680
678154


1186060


219404


284947


1014035


563479


284952


776475


235804


1169693


Status:  1690
932138
1153322


915756


842029


80174
530733


1177908


555317


1153334


350518


Status:  1700
842040


170301


424263


366926


1161550


932181


891222


317789


1153375


407906


Status:  1710
Status:  1710
260464


833906


1153395


555389


506239
1153408


96640


399746


186755


1177993


Status:  1720
211350


1153433


1153435


1014179


506277


1153446


1153447


1120680


252326
891307
Status:  1730
219569


1153457


211380


227766


1161655


620985
1153467


793022


1030593


31179


Status:  1740
1055180


940495


88529


129489
1014228


88537
702938


752094


154084


1104361


Status:  1750
858607
784880
596467


1071604


63989


760312
1153533


1169918


301567


293378


Status:  1760
571909


1194503
162315


309772
1153554


891410


1169943


145959


940584


227883


Status:  1770
195115
1161778


678460


1169980


47682


629324


211533


1170004


1170024


1071721


Status:  1780
277109


96889


260738


588418


637576


350859


457357


1178254


727695


932494
Status:  1790
285325


850578


162450


785050


211613


88735


440992


965279


1129126


1161898


Status:  1800
408236


776879


39600


1153718


514745


1178300
1096381


785087


1006276


285387


Status:  1810
490188


1186508


260818


1153749


1194709
121562


1006308


482034


1153785


506620


Status:  1820
932610


1006339


1022724


1178374


1153800


211721


924426


523027


523029


506646


Status:  1830
1153815


1030938


1153825


1112866


965411


777001


768813


1170226


154419


56123


Status:  1840
129852


686911


129858


1162053


891719


1153864
97097


129866
1194828


351057


Status:  1850
1162069


129878


1153878


1170267


1153901


1039214


1178478


1153905
1039218
1170306


Status:  1860
596867
1178500


1088390


875399


744330


236426
457617


310184


For 310184 cannot find leaderboard table file: table_02.csv
For 310184 cannot find leaderboard table file: table_03.csv
For 310184 cannot find leaderboard table file: table_04.csv
293804


1047471


Status:  1870
Status:  1870
146355


1153973


1194934


1113015
1186745


768956


1022922


1178578


826329


1162203


Status:  1880
695268
1154028


1047533


1047534


1178611
1154044


908285


408575


711680


654336


Status:  1890
457731


1129478


343049


900106


1154057


48137
654350


1154064


400401


457746


Status:  1900
187414


1154074


1154076
211996


1162273


433195


154668


252982


236599


990263
Status:  1910
949306


982074


449596


982078


1154111


1113154
1162306


973892


457815


146520


Status:  1920
498782


760932


818278


1154153


1178733


1154158
769140


842869


72821


1186932


Status:  1930
752764


375933
580738
941188


23685


965766


1072265


834703


720019


449684


Status:  1940
1178773


1178774


760983


867482


416925


1170594


1162407


539819


769199


1137843


Status:  1950
Status:  1950
Status:  1950
1154236


1178813


163018


130255


408784


244944


1154256


548061


1178857


310508
Status:  1960
1170679


1178872


15621


40198


818442


89355
376078
728338


97556


580890


Status:  1970
1170717


580894


1187103


294177


220450


1154339


89381


1105195


777520


548147


Status:  1980
1187131


425275


1162557


163135


294208


1170753


1162577


1006931
1162585


1121631


Status:  1990
1023329
1154406


154986


769391


163191
15740


507267


81294


884110


245137
Status:  2000
15761


1154450
302486


1154456


1154458
941467


1154463
728480


1080741


417197


Status:  2010
1154486


236992


802248


982473


1179089


171477


1179102


376295


1170921


744942


Status:  2020
1154547
1154560


1170945


409090


1154563


212487
179723
1179148


777744


917016


Status:  2030
777755


1154588


146973


1162782


728607


196133


1162790


1023528


589353


1162793


Status:  2040
1154610


1154614


1171000


114233


278074


1154619


1154626


1007174


179786


1171021


Status:  2050
548432


1154641


1179234


1154663


278120


89706


720491
630381


1064559


425587


Status:  2060
1171061
654971


999037
327296


228996


1007237


695947


212619


1179277


1154702


Status:  2070
1154714


1121947


1187486


417439


818850


999075


671398


310953


442033


1154741


Status:  2080
556729


1154749


204480


728771


540356


974533
884428
466638


949975


1154775
Status:  2090
1171168


409316


278247


1048301


1113841


368371


138995


65269
1187574


130809


Status:  2100
163578


630527
32513


311042


1163015


1171208
1081099


384786


352021


1154839


Status:  2110
229145


761629


1171241


1179437


1171248


327474


1138484


147256


991038


1097536


Status:  2120
Status:  2120
Status:  2120
Status:  2120
925518
892759


1154904
1179480


286557


1007456


1195875


1154917
65398


352134


Status:  2130
532358


1163147


630669


917397


655259


909215


1171367
393129


171945


212914


Status:  2140
933811


516027


909246


1089480


319439
1114066


532440
1163226


1138653


1155037


Status:  2150
1155038


720871


327659
1097708


819181


1130480


1187829


65533


In [46]:
status_count, len(unextracted_papers), 2158+175

(2158, 175, 2333)

In [54]:
len(leaderboard_refs), len(leaderboard_table_refs), 2153+175

(1668, 2153, 2328)

In [48]:
list(leaderboard_refs.items())[0:2]

[(1179651,
  {'refs': {'ldb': ["<ref id='bib-bib5'>5</ref>",
     "<ref id='bib-bib34'>34</ref>",
     "<ref id='bib-bib43'>43</ref>",
     "<ref id='bib-bib31'>31</ref>",
     "<ref id='bib-bib32'>32</ref>"],
    'abl': []},
   'count': 5}),
 (1179653, {'refs': {'ldb': [], 'abl': []}, 'count': 0})]

In [50]:
list(leaderboard_table_refs.items())[0:2]

[(1179651,
  {'tcount': [9, 6, 1, 2],
   'table_labels': [('irrelevant',),
    ('irrelevant',),
    ('leaderboard',),
    ('leaderboard',),
    ('ablation',),
    ('leaderboard',),
    ('leaderboard',),
    ('leaderboard',),
    ('leaderboard',)]}),
 (1179653,
  {'tcount': [9, 4, 5, 0],
   'table_labels': [('leaderboard',),
    ('leaderboard',),
    ('ablation',),
    ('ablation',),
    ('ablation',),
    ('leaderboard',),
    ('ablation',),
    ('ablation',),
    ('leaderboard',)]})]

# ldb table present but no references

In [55]:
zero_refs = 0

for k in leaderboard_table_refs:
    if not k in unextracted_papers and (leaderboard_table_refs[k]['tcount'][1] + leaderboard_table_refs[k]['tcount'][2]) == 0:
        zero_refs += 1
        print("No refs: ", k)
    if (leaderboard_table_refs[k]['tcount'][1] + leaderboard_table_refs[k]['tcount'][2]) > 0 and not k in leaderboard_refs:
        print("Is there a mistake somewhere?:", k)

No refs:  598027
No refs:  557076
No refs:  647191
No refs:  327741
No refs:  1171539
No refs:  540762
No refs:  868481
No refs:  368771
No refs:  1147022
No refs:  163988
No refs:  1155227
No refs:  860339
No refs:  106720
No refs:  671972
No refs:  1155304
No refs:  1147188
No refs:  1048899
No refs:  942483
No refs:  328106
No refs:  205248
No refs:  360922
No refs:  1163743
No refs:  1122786
No refs:  475630
No refs:  1163769
No refs:  1155584
No refs:  705037
No refs:  696851
No refs:  1040928
No refs:  213553
No refs:  270911
No refs:  639555
No refs:  180805
No refs:  287334
No refs:  107113
No refs:  279176
No refs:  819849
No refs:  459424
No refs:  1155754
No refs:  574126
No refs:  901811
No refs:  1082054
No refs:  1155803
No refs:  262920
No refs:  353047
No refs:  992046
No refs:  279371
No refs:  181085
No refs:  820065
No refs:  713621
No refs:  1098664
No refs:  730027
No refs:  1147846
No refs:  754630
No refs:  943067
No refs:  533486
No refs:  582693
No refs:  99367

In [57]:
print(zero_refs, 1668 + 485, 1668 + 485 + 175)

485 2153 2328


In [63]:
nk = 630527




leaderboard_table_refs[nk], iclr_ref_arxiv_map[nk]

({'tcount': [0, 0, 0, 0], 'table_labels': []}, '1506.08700')

## One weird thing for these papers is that there are no leaderboard/ablation tables in around 485 papers out of 2153 papers (2153 = 2328 - 175)

## Counting papers that have one ldb/abl tables and at least one reference

In [75]:
zero_refs_but_ldb = []
not_in_ldb_refs = []
succ = 0

for k in leaderboard_table_refs:
    if not k in unextracted_papers and (leaderboard_table_refs[k]['tcount'][1] + leaderboard_table_refs[k]['tcount'][2]) >= 0:
        if k in leaderboard_refs:
            if 'count' in leaderboard_refs and leaderboard_refs[k]['count'] <= 0:
                zero_refs_but_ldb.append(k)
            else:
                succ+=1
        else:
            not_in_ldb_refs.append(k)
    else:
        msg = "We dont care about these. We have looked into these above already. These are 175 + 485 papers for each of the above conditions resp."

In [82]:
len(leaderboard_refs), len(zero_refs_but_ldb), len(not_in_ldb_refs), succ

(1668, 0, 485, 1668)

So basically if the key is not in leaderboard_refs, then it means that no reference was extracted from the ldb/abl tables

In [80]:
len(leaderboard_refs), list(leaderboard_refs.items())[0:2]

(1668,
 [(1179651,
   {'refs': {'ldb': ["<ref id='bib-bib5'>5</ref>",
      "<ref id='bib-bib34'>34</ref>",
      "<ref id='bib-bib43'>43</ref>",
      "<ref id='bib-bib31'>31</ref>",
      "<ref id='bib-bib32'>32</ref>"],
     'abl': []},
    'count': 5}),
  (1179653, {'refs': {'ldb': [], 'abl': []}, 'count': 0})])

In [83]:
useful_nodes = []
non_useful = []

for k in leaderboard_refs:
    if "count" in leaderboard_refs[k] and leaderboard_refs[k]["count"] > 0:
        useful_nodes.append(k)
    else:
        non_useful.append(k)
        
print(len(useful_nodes), len(non_useful))

682 986


In [84]:
useful_nodes[0:2], non_useful[0:5]

([1179651, 229423], [1179653, 1163279, 983062, 557082, 172080])

### Ins0

In [85]:
nk = 1179653
iclr_ref_arxiv_map[nk], leaderboard_table_refs[nk]

('1905.02249',
 {'tcount': [9, 4, 5, 0],
  'table_labels': [('leaderboard',),
   ('leaderboard',),
   ('ablation',),
   ('ablation',),
   ('ablation',),
   ('leaderboard',),
   ('ablation',),
   ('ablation',),
   ('leaderboard',)]})

In [89]:
paper = pc.get_by_id(iclr_ref_arxiv_map[nk])

In [93]:
paper.tables[0].df.loc[1,0]

(<axcell.data.table.Table at 0x7f0f4b9f04e0>,
 Cell(value='Mean Teacher tarvainen2017weight', raw_value='Mean Teacher tarvainen2017weight', gold_tags='', refs=[], layout='border-t align-left header'))

# Ins1

In [97]:
nk = 557082
iclr_ref_arxiv_map[nk], leaderboard_table_refs[nk]

('1411.2005', {'tcount': [1, 1, 0, 0], 'table_labels': [('leaderboard',)]})

In [98]:
paper = pc.get_by_id(iclr_ref_arxiv_map[nk])

In [99]:
paper.tables[0]